src/services/analytics/core.ts

```diff
@@ -0,0 +1,238 @@
+  // Values: slow-2g, 2g, 3g, 4g
+  if (effectiveType) {
+    context.effectiveType = effectiveType;
+  }
+
+  return context;
+};
+
+/**
+ * Dispatches event to Amplitude without throttling.
+ * Respects user analytics preferences.
+ */
+const dispatchUnthrottled = (
+  event: AnalyticsEventName,
+  props?: AnalyticsProps,
+): void => {
+  const { isEnabled } = useAnalyticsStore.getState();
+
+  const eventData = ANALYTICS_CONFIG.INCLUDE_PLATFORM_CONTEXT
+    ? { ...buildCommonContext(), ...props }
+    : props;
+
+  // Always add to debug buffer for development visibility
+  addToRecentEvents(event, eventData);
+
+  // Only send to Amplitude if conditions are met
+  if (!isEnabled) {
+    logger.debug(DEBUG_CONFIG.LOG_PREFIX, `Skipping disabled event: ${event}`);
+
+    return;
+  }
+
+  if (!AMPLITUDE_API_KEY) {
+    logger.debug(
+      DEBUG_CONFIG.LOG_PREFIX,
+      `Skipping event due to missing API key: ${event}`,
+    );
+
+    return;
+  }
+
+  if (!hasInitialised) {
+    logger.warn(
+      DEBUG_CONFIG.LOG_PREFIX,
+      `Analytics not initialized, skipping: ${event}`,
+    );
+    return;
+  }
+
+  try {
+    amplitude.track(event, eventData);
+
+    logger.debug(DEBUG_CONFIG.LOG_PREFIX, `Event sent to Amplitude: ${event}`);
+  } catch (error) {
+    logger.error(DEBUG_CONFIG.LOG_PREFIX, `Failed to track event: ${event}`, {
+      error,
+      props,
+    });
+  }
+};
+
+/**
+ * Throttled event dispatcher to prevent duplicate events.
+ * Uses trailing execution to give time for analytics preferences to load.
+ * We use memoize to create a separate throttled function for each event name.
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const getThrottledDispatcher = memoize((_eventName: AnalyticsEventName) =>
+  throttle(dispatchUnthrottled, TIMING.THROTTLE_DELAY_MS, {
+    leading: false,
+    trailing: true,
+  }),
+);
+
+/**
+ * Main event tracking function.
+ * Uses throttling by default to prevent duplicate events.
+ */
+export const track = (
+  event: AnalyticsEventName,
+  props?: AnalyticsProps,
+): void => {
+  if (ANALYTICS_CONFIG.THROTTLE_DUPLICATE_EVENTS) {
+    // Get a dispatcher throttled for this specific event name
+    const throttledDispatch = getThrottledDispatcher(event);
+
+    throttledDispatch(event, props);
+  } else {
+    dispatchUnthrottled(event, props);
+  }
+};
+
+// -----------------------------------------------------------------------------
+// APP LIFECYCLE
+// -----------------------------------------------------------------------------
+
+export const trackAppOpened = (): void => {
+  track(AnalyticsEvent.APP_OPENED);
+};
+
+// -----------------------------------------------------------------------------
+// STORE SUBSCRIPTION (Must be after track function is defined)
+// -----------------------------------------------------------------------------
+
+// Set up analytics store subscription to handle user toggling analytics
+useAnalyticsStore.subscribe((state) => {
```

Comment:
this is great, I didn't know we could easily subscribe to store changes like that